nexus/test-utils/src/lib.rs

```diff
@@ -290,11 +293,40 @@ impl<'a, N: NexusServer> ControlPlaneTestContextBuilder<'a, N> {
             external_dns_zone_name: None,
             external_dns: None,
             internal_dns: None,
+            dns_config: None,
             silo_name: None,
             user_name: None,
         }
     }
 
+    pub async fn init_with_steps(
+        &mut self,
+        steps: Vec<(
+            &str,
+            Box<dyn for<'b> FnOnce(&'b mut Self) -> BoxFuture<'b, ()>>,
```

Comment:
Yay for no async closures :(

Reply:
I know, also been suffering from this recently. Sucks to need to lean on box so heavily to get this all done.